src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationFileChangeDetector.cs

```diff
@@ -92,11 +92,23 @@ public async Task StartAsync(string workspaceDirectory, CancellationToken cancel
 
                 if (args.OldFullPath.EndsWith(LanguageServerConstants.ProjectConfigurationFile, FilePathComparison.Instance))
                 {
+                    if (args.FullPath.EndsWith(".old"))
+                    {
+                        // The file is actually getting modified, do nothing
+                        return;
+                    }
+
                     // Renaming from project.razor.json to something else. Just remove the configuration file.
                     FileSystemWatcher_ProjectConfigurationFileEvent_Background(args.OldFullPath, RazorFileChangeKind.Removed);
                 }
                 else if (args.FullPath.EndsWith(LanguageServerConstants.ProjectConfigurationFile, FilePathComparison.Instance))
                 {
+                    if (args.OldFullPath.EndsWith(".temp"))
+                    {
+                        // The file is actually getting modified, a changed event will fire to take care of it.
```

Comment:
O_o why does an extra change event fire? This seems fragile

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher?view=netframework-4.8#events-and-buffer-sizes. Basically because FileSystemWatcher looks at the system events and even something like a simple move might actually be 3 operations (Delete old one, create new one, modify last updated time).

Reply:
See the "might actually be 3 operations" is the concerning piece I have. This logic will run on mac/linux and windows all which have different file system impls. Basically I'm concerned about the assumption that a changed event will absolutely fire after a rename.